app/middleware/receiveChartOptions.js

```diff
@@ -26,15 +26,21 @@ export default function receiveChartType({ getState }) {
     // Clone received options
     let nextOpts = update({}, { $set: action.data });
 
-    // If we have nvd3MultiSeries dataFormat but colors not yet applied
+    /**
+     * Determine from store if we need to apply colors to data series.
+     * True when dataFormat is nvd3MultiSeries and
+     * action comes from manual user change OR not all series have a color already
+     */
     function _shouldApplyColorsToData() {
       let shouldApply;
       try {
-        // If dataFormat is nvd3MultiSeries, check first and last series
-        // to see if colors have already been applied
+        const allSeriesHaveColor = getState().chartData.reduce((acc, series) =>
```

Comment:
I'm kinda curious what performance difference between doing this and `chartData.filter(({ color }) => 'undefined' === typeof color).length > 0` would be...